core/services/feeds/orm.go

```diff
@@ -564,20 +564,36 @@ SELECT exists (
 // DeleteProposal performs a soft delete of the job proposal by setting the status to deleted and
 // update the status to deleted
 func (o *orm) DeleteProposal(id int64, qopts ...pg.QOpt) error {
+	// Get the latest spec for the proposal. We should set pending update to true only if the
+	// latest proposal is approved so the running jobs can be reminded to be deleted.
 	stmt := `
+	SELECT id, definition, version, status, job_proposal_id, status_updated_at, created_at, updated_at
+FROM job_proposal_specs
+WHERE (job_proposal_id, version) IN
+(
+	SELECT job_proposal_id, MAX(version)
+	FROM job_proposal_specs
+	GROUP BY job_proposal_id
+)
+AND job_proposal_id = $1
+`
+
+	var spec JobProposalSpec
+	err := o.q.WithOpts(qopts...).Get(&spec, stmt, id)
+	if err != nil {
+		return err
+	}
+
+	pendingUpdate := spec.Status == SpecStatusApproved
```

Comment:
```suggestion
        // Set pending update to true only if the latest proposal is approved.
        // so the running jobs can be reminded to be deleted.
	pendingUpdate := spec.Status == SpecStatusApproved
```

what does `so the running jobs can be reminded to be deleted` mean? How does this statement relate to the `pendingUpdate` status of the latest spec?

Reply:
When pending update is set to true, the UI uses this to display bubbles that action is needed. If a proposal is deleted and there are no approved jobs for the proposal, there is no action needed. If there are running jobs, the node op should cancel those jobs.